probe-rs/src/session.rs

```diff
@@ -377,7 +377,12 @@ impl Session {
 
                     JtagInterface::Riscv(state)
                 }
-                Architecture::Xtensa => JtagInterface::Xtensa(XtensaDebugInterfaceState::default()),
+                Architecture::Xtensa => {
+                    // Workaround for CMSIS-DAP probe not scanning in `attach_to_unspecified`.
+                    let mut state = XtensaDebugInterfaceState::default();
+                    probe.try_get_xtensa_interface(&mut state)?;
```

Comment:
I am not too fond of this. Couldn't we make/use a more explicit function/method to get the desired effect?

Reply:
We can change Probe::scan_chain() to scan the chain when empty. The problem we have is that the arm debug interface works differently from the other JTAGs - ARM prefers the user-specified data, the others just use it to double-check what they scan. I think untangling that is out of scope here.